packages/jest-fake-timers/src/jestFakeTimers.ts

```diff
@@ -368,6 +369,15 @@ export default class FakeTimers<TimerRef> {
       // @ts-ignore TODO: figure out better typings here
       this._moduleMocker.fn().mockImplementation(impl);
 
+    const promisifiedFakeSetTimeout = fn(this._fakeSetTimeout.bind(this));
+    promisifiedFakeSetTimeout[util.promisify.custom] = (
+      delay?: number,
+      arg?: any,
+    ) =>
+      new Promise(resolve =>
+        promisifiedFakeSetTimeout(() => {}, delay, resolve(arg)),
```

Comment:
Shouldn't `resolve` be called in the function passed to the setTimeout function as a first argument? I don't think we need the third argument here.

Reply:
Hi, @jeysal. Thanks for the review.

I've implemented `setTimeout[util.promisify.custom]` like the original one implemented on node. It doesn't call the callback passed through it. Moreover, the first argument is reserved to the delay. You can see this in this [repl](https://repl.it/@lourenci/FantasticSquigglyAngle).
If we resolve the first argument, the callback will be called and the delay will be the second argument. This is not how the node's implementation work.

Reply:
If we don't care about the third parameter, we can do something like that:
`
    promisifiedFakeSetTimeout[util.promisify.custom] = (
      delay: number,
    ) =>
      new Promise(resolve =>
        promisifiedFakeSetTimeout(resolve, delay),
      );
`

Reply:
Sorry for the delay in getting back to you.

> I've implemented `setTimeout[util.promisify.custom]` like the original one implemented on node.

Really? It does not look like it resembles [Node's implementation](https://github.com/nodejs/node/blob/8709b905833d2078eaeec65a5e6dbce16096ae00/lib/timers.js#L148-L153).

> If we don't care about the third parameter

I think we do care! But your implementation does not look correct, calling `resolve` in the function passed to the `Promise` constructor will resolve the Promise immediately.

Reply:
Of course, my bad. I've done the changes.